internal/setup/setup_test.go

```diff
@@ -2240,3 +2240,53 @@ func (s *S) TestParseSliceKey(c *C) {
 		c.Assert(key, DeepEquals, test.expected)
 	}
 }
+
+var sampleText = "sample"
+
+var yamlPathTests = []struct {
+	path1, path2 *setup.YAMLPath
+	areSame      bool
+}{{
+	path1: &setup.YAMLPath{
+		Text:    &sampleText,
+		Mutable: true,
+		Mode:    0644,
+		Arch:    setup.YAMLArch{List: []string{"amd64", "arm64"}},
+	},
+	path2: &setup.YAMLPath{
+		Text:    &sampleText,
+		Mutable: true,
+		Mode:    0644,
+		Arch:    setup.YAMLArch{List: []string{"amd64", "arm64"}},
+	},
+	areSame: true,
+}, {
+	path1:   &setup.YAMLPath{Dir: true, Mode: 0755},
+	path2:   &setup.YAMLPath{Dir: true, Mode: 0755},
+	areSame: true,
+}, {
+	path1:   &setup.YAMLPath{Symlink: "foo"},
+	path2:   &setup.YAMLPath{Symlink: "foo"},
+	areSame: true,
+}, {
+	path1:   &setup.YAMLPath{Generate: setup.GenerateManifest},
+	path2:   &setup.YAMLPath{Generate: setup.GenerateManifest},
+	areSame: true,
+}, {
+	// "until" is not checked.
```

Comment:
I think once we have to include comments in the tests it is a good idea to add a `summary: string` attribute.

Reply:
Alright, added summaries in 1810c74.